### PR TITLE
Referendum decision period quantization

### PR DESCRIPTION
**Pull request by muharem**
Right now if `AlarmInterval` set to > 1, the time for alarm will be calculated to the past, and scheduler will fail.

This PR fixes it.

### CODE REVIEW

File: frame/referenda/src/lib.rs [hunk 1]
```diff
@@ -754,9 +759,6 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		call: BoundedCallOf<T, I>,
 		when: T::BlockNumber,
 	) -> Option<(T::BlockNumber, ScheduleAddressOf<T, I>)> {
-		let alarm_interval = T::AlarmInterval::get().max(One::one());
-		let when = when.saturating_add(alarm_interval).saturating_sub(One::one()) /
-			(alarm_interval.saturating_mul(alarm_interval)).max(One::one());
```
Comment (muharem) [inline frame/referenda/src/lib.rs:759]: its probably supposed to be calculated from the duration

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2103277

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2103276

File: frame/referenda/src/lib.rs [hunk 2]
```diff
@@ -195,12 +195,17 @@ pub mod pallet {
...
+		/// Quantization level for the referendum wakeup scheduler during the decision stage. A
+		/// higher number will result in fewer storage reads/writes needed for smaller voters, but
+		/// also result in delays to the automatic referendum status changes.
 		#[pallet::constant]
 		type AlarmInterval: Get<Self::BlockNumber>;
 
+		/// Minimum alarm period for the referendum wakeup scheduler during the decision stage.
+		/// Prevents wake up calls from being too frequent as a result of the quantization.
+		#[pallet::constant]
+		type MinAlarmPeriod: Get<Self::BlockNumber>;
```
Comment (gavofyork) [inline frame/referenda/src/lib.rs:207]: What's the actual use-case for this?

**Comment by gavofyork** [comment]
Please add a test.

**Comment by gavofyork** [comment]
Please see https://github.com/paritytech/substrate/pull/12815 and check if it passes your test.

**Comment by muharem** [comment]
> Please see #12815 and check if it passes your test.

got it wrong actually in this PR, please check an addition to your pr, test and note_one_fewer_deciding uses set_alarm  https://github.com/paritytech/substrate/pull/12818

**Comment by muharem** [comment]
Closing in favor of this pr https://github.com/paritytech/substrate/pull/12815

**Pull request closed without merging**